### PR TITLE
Add support for 1.20.6

### PR DESCRIPTION
**Pull request by Gilly7CE**
This PR introduces a support change for Minecraft (MC) 1.20.6. This directly bumps the `1.20.x` branch from 1.20.2 to 1.20.6 (including 1.20.4 commits which were originally missing).

Full list of changes:
- Update Gradle to 8.10 and Loom to 1.7-Snapshot
- Update MC support to 1.20.6
- Replace `EndFrameBlock_EyeToggleMixin` with `EnderEyeItem_EyeToggleMixin` following removal of deprecated methods in `Block` code
- Refactor spectator portal rule code to not use networking sync between server and client but two mixins for server and client which executes the same logic, like in vanilla. Networking changes in 1.20.5 caused a breaking change here and it would be too complicated to keep around for a single use case. The new implementation is likely more future proof.

### CODE REVIEW

File: src/main/java/carpetaddonsnotfound/spectatorplayersuseportals/SpectatorPlayersUsePortalsRule.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Box;
+import net.minecraft.world.World;
+
+public class SpectatorPlayersUsePortalsRule {
+  public static void movePlayerInSpectator(PlayerEntity playerEntity) {
+    final boolean allowVehicles = false;
```
Comment (Gilly7CE) [inline src/main/java/carpetaddonsnotfound/spectatorplayersuseportals/SpectatorPlayersUsePortalsRule.java:15]: Remove this variable as it is not needed for 1.20.6

File: src/main/resources/fabric.mod.json [hunk 2]
```diff
@@ -17,10 +17,10 @@
   },
   "mixins": ["carpet-addons-not-found.mixins.json"],
   "depends": {
-    "fabricloader": ">=0.14.22",
-    "minecraft": "1.20.2",
-    "java": ">=17",
-    "carpet": ">=1.4.112"
+    "fabricloader": ">=0.16.3",
+    "minecraft": ">=1.21.1",
```
Comment (Gilly7CE) [inline src/main/resources/fabric.mod.json:21]:
```suggestion
    "minecraft": ">=1.20.6",
```

Copied this wrong, whoops

Comment (Gilly7CE) [reply src/main/resources/fabric.mod.json]: Resolved

File: src/main/java/carpetaddonsnotfound/spectatorplayersuseportals/SpectatorPlayersUsePortalsRule.java (hunk 1 above)
Comment (Gilly7CE) [reply src/main/java/carpetaddonsnotfound/spectatorplayersuseportals/SpectatorPlayersUsePortalsRule.java]: Removed

**Review by connorslade — APPROVED** [review]

**Pull request merged**